src/App.tsx

```diff
@@ -1,18 +1,27 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
-
 export const App: React.FC = () => {
+  const arrNums = getNumbers(1, 42);
+
+  const [perPage, setPerPage] = useState(10);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [items, setItems] = useState<string[]>([]);
+
+  useEffect(() => {
+    setItems(([...arrNums]).splice((currentPage - 1) * perPage, perPage).map(n => `Item ${n}`));
+  }, [currentPage, perPage]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage}
+         (items ${currentPage === 1 ? 1 : ((currentPage - 1) * perPage) + 1}
+         - ${(currentPage * perPage) > 42 ? 42 : currentPage * perPage} of 42)`}
```

Comment:
In such long expressions better use variables.
```suggestion
        {`Page ${currentPage} (${fromItemValue} - ${toItemValue} of ${arrNums.length})`}
```